nestkernel/stimulating_backend_mpi.h

```diff
@@ -0,0 +1,149 @@
+/*
+ *  stimulating_backend_mpi.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef STIMULATING_BACKEND_MPI_H
+#define STIMULATING_BACKEND_MPI_H
+
+#include "stimulating_backend.h"
+#include "nest_types.h"
+#include "nest_time.h"
+#include <set>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <mpi.h>
+
+/* BeginDocumentation
+
+Read data from MPI
+##################
+
+When an stimulating device is to be updated at the beginning of each step run, the 'mpi' backend
+communicates with an external source via MPI to get the update.
+
+Communication Protocol: (value,number,type,source/destination,tag)
++++++++++++++++++++++++
+1) Connection of MPI port include in one file ( path+label+id+.txt )
+2) Send at each beginning run of Nest (true, 1, CXX_BOOL, 0, 0)
+3) Send the id of the device to update (id_device, 1, INT, 0, 0)
+3) Receive shape of the data (shape, 1, INT, 0, 0)
+4) Receive the data for updating the device (data, shape, DOUBLE, 0, 0)
+5) Send at each ending of the run (true, 1, CXX_BOOL, 0, 1)
+6) Send at this en of the simulation (true, 1, CXX_BOOL, 0, 2)
+
+
+@author Lionel Kusch and Sandra Diaz
+@ingroup NESTio
+
+EndDocumentation */
```

Comment:
Please make this into a user documentation block as it is for example in [`recording_backend_ascii.h`](https://github.com/nest/nest-simulator/blob/master/nestkernel/recording_backend_ascii.h#L31-L139) so it can be properly picked up by our Sphinx pipeline.

Also, please add some information to the [Guide for stimulating the network](https://github.com/nest/nest-simulator/blob/master/doc/userdoc/guides/stimulating_the_network.rst) that explains how to use stimulation backends from a user perspective.